File: Buoi2/Julia/Selection.jl
# Hàm Selection Sort
function selection_sort(arr)
    n = length(arr)
    for i in 1:n-1
        # Tìm chỉ số của phần tử nhỏ nhất trong [i..n-1]
        min_index = i
        for j in i+1:n
            if arr[j] < arr[min_index]
                min_index = j
            end
        end
        # Hoán đổi nếu min_index khác i
        if min_index != i
            arr[i], arr[min_index] = arr[min_index], arr[i]
        end
    end
end

arr = [5, 2, 9, 1, 5]
selection_sort(arr)
println("Kết quả sau Selection Sort:", arr)

File: Buoi2/C++/Selection.cpp
#include <iostream>
#include <vector>
using namespace std;

// Hàm Selection Sort
void selectionSort(vector<int>& arr) {
    int n = arr.size();
    for (int i = 0; i < n - 1; i++) {
        // Tìm chỉ số của phần tử nhỏ nhất trong [i..n-1]
        int min_index = i;
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[min_index]) {
                min_index = j;
            }
        }
        // Hoán đổi nếu min_index khác i
        if (min_index != i) {
            swap(arr[i], arr[min_index]);
        }
    }
}

int main() {
    vector<int> arr = {5, 2, 9, 1, 5};
    selectionSort(arr);
    cout << "Ket qua sau Selection Sort: ";
    for (int x : arr) {
        cout << x << " ";
    }
    cout << endl;
    return 0;
}
